test/lib/rules/no-async-operation.js

```diff
@@ -87,3 +84,78 @@ ruleTester.run('no-async-operations', rule, {
         },
     ],
 });
+
+testTypeScript('no-async-operation', {
+    valid: [
+        {
+            code: 'obj.setTimeout();',
+        },
+        {
+            code: `
+                const obj: object = {}; 
+                obj.setTimeout();
+            `,
+        },
+        {
+            code: `
+                const setTimeout: Function = () => {}; 
+                setTimeout();
+            `,
+        },
+        {
+            code: `
+                const setTimeout: Function = () => {}; 
+
+                function foo() {
+                    setTimeout();
+                }
+            `,
+        },
+    ],
+    invalid: [
+        {
+            code: `
+                setTimeout() satisfies number;
+                setInterval() satisfies number;
+                requestAnimationFrame() satisfies number;
+            `,
+            errors: [
+                { message: 'Restricted async operation "setTimeout"' },
+                { message: 'Restricted async operation "setInterval"' },
+                { message: 'Restricted async operation "requestAnimationFrame"' },
+            ],
+        },
+        // {
+        //     code: `
+        //         (window as any).setTimeout();
+        //         (window as any).setInterval();
+        //         (window as any).requestAnimationFrame();
+        //     `,
+        //     errors: [
+        //         { message: 'Restricted async operation "setTimeout"' },
+        //         { message: 'Restricted async operation "setInterval"' },
+        //         { message: 'Restricted async operation "requestAnimationFrame"' },
+        //     ],
+        // },
+        // {
+        //     code: `
+        //         (window as any)["setTimeout"]();
+        //         (window as any)["setInterval"]();
+        //         (window as any)["requestAnimationFrame"]();
+        //     `,
+        //     errors: [
+        //         { message: 'Restricted async operation "setTimeout"' },
+        //         { message: 'Restricted async operation "setInterval"' },
+        //         { message: 'Restricted async operation "requestAnimationFrame"' },
+        //     ],
+        // },
```

Comment:
Is this code needed?

Reply:
This is a case that _should_ be reported as invalid, but isn't because the type assertions bypass whatever checks we do. Once #154 is fixed we should be able to un-comment these lines and have the tests pass.

(It would be nice if `RuleTester` supported `skip`, but it doesn't.)